redhat-cop/infra-ansible
# Add conditional for credentials managed by tower

### What does this PR do?

Add conditional for Tower-managed credentials during cleanup which will skip any credential with `"managed_by_tower": true`

### How should this be tested?

This really only affects Tower >= 3.8. A default has been added so that Tower 3.7 instances are not broken by the change.

- Add a couple of new credentials with `playbooks/ansible/tower/configure-ansible-tower.yml`
- Remove only one of the credentials and confirm there are no errors from Tower and that the correct credentials exist alongside the Tower-managed `Ansible Galaxy` credential

### Other Relevant info, PRs, etc.

Just a note that `credential_types` also have the `managed_by_tower` field, however we are not currently cleaning up undeclared credential_types how we are with credentials. None of the other "Default" objects we might want to clean up (such as organizations and job templates) currently have this field.

### People to notify

cc: @redhat-cop/infra-ansible
